src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+	end
+	
+	if CUSTOM_DIMENSIONS.DIMENSION_02 then
+		GameAnalytics:configureAvailableCustomDimensions02(CUSTOM_DIMENSIONS.DIMENSION_02)
+	end
+	
+	if CUSTOM_DIMENSIONS.DIMENSION_03 then
+		GameAnalytics:configureAvailableCustomDimensions03(CUSTOM_DIMENSIONS.DIMENSION_03)
+	end
+	
+	GameAnalytics:configureAvailableResourceCurrencies(CURRENCIES)
+	GameAnalytics:configureAvailableResourceItemTypes(RESOURCE_EVENT_TYPES)
+	GameAnalytics:configureAvailableGamepasses(GAMEPASS_IDS)
+	
+	GameAnalytics:setEnabledInfoLog(false)
+	GameAnalytics:setEnabledVerboseLog(false)
+	GameAnalytics:setEnabledDebugLog(false)
+	
+	GameAnalytics:setEnabledAutomaticSendBusinessEvents(false)
+	GameAnalytics:setEnabledReportErrors(false)
+	
+	GameAnalytics:initServer(GAME_KEY, SECRET_KEY)
+end
+
+function AnalyticsService:KnitStart()
+	self.Client.LogEvent:Connect(function(player: Player, data: {
+		event: string,
+		value: number?
+	})
+		self:LogPlayerEvent({
+			userId = player.UserId,
+			event = data.event,
+			value = data.value
+		})
+	end)
+	
+	MarketplaceService.PromptBundlePurchaseFinished:Connect(
+		function(player: Player, bundleId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Bundle",
+				id = bundleId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptGamePassPurchaseFinished:Connect(
+		function(player: Player, gamePassId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "GamePass",
+				id = gamePassId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	if LOG_DEV_PRODUCT_PURCHASES then
+		MarketplaceService.PromptProductPurchaseFinished:Connect(
+			function(player: Player, productId: number, purchased: boolean)
+				if not purchased then
+					return
+				end
+				
+				self:LogMarketplacePurchase({
+					userId = player.UserId,
+					itemType = "Product",
+					id = productId,
+					cartType = "PromptPurchase",
+				})
+			end
+		)
+	end
+	
+	MarketplaceService.PromptPurchaseFinished:Connect(
+		function(player: Player, assetId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Asset",
+				id = assetId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptSubscriptionPurchaseFinished:Connect(
+		function(player: Player, subscriptionId: string, didTryPurchasing: boolean)
+			if not didTryPurchasing then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Subscription",
+				id = subscriptionId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+end
+
+--[=[
+	Used to track player events (example: player killed an enemy, player completed a mission, etc.)
```

Comment:
PlayerEvent type is defined earlier in the script; I would add a reference to that type here (Eg. [PlayerEvent]); that way Moonwave points to that type/the information about it.

Add param tag to the moonwave block.
@param data PlayerEvent -- PlayerEvent type